pallets/dca/src/lib.rs

```diff
@@ -0,0 +1,971 @@
+		///The next execution block number is not in the future
+		BlockNumberIsNotInFuture,
+		///Price change from oracle data is bigger than max allowed
+		PriceChangeIsBiggerThanMaxAllowed,
+		///Error occurred when calculating price
+		CalculatingPriceError,
+		///The total amount to be reserved is smaller than min budget
+		TotalAmountIsSmallerThanMinBudget,
+		///The budget is too low for executing one DCA
+		BudgetTooLow,
+		///There is no free block found to plan DCA execution
+		NoFreeBlockFound,
+		///The DCA schedule has been manually terminated
+		ManuallyTerminated,
+		///Max number of retries reached for schedule
+		MaxRetryReached,
+		///The trade limit has been reached, leading to retry
+		TradeLimitReached,
+		///The route to execute the trade on is not specified
+		RouteNotSpecified,
+		///No parent hash has been found from relay chain
+		NoParentHashFound,
+		///Error that should not really happen only in case of invalid state of the schedule storage entries
+		InvalidState,
+	}
+
+	/// Id sequencer for schedules
+	#[pallet::storage]
+	#[pallet::getter(fn next_schedule_id)]
+	pub type ScheduleIdSequencer<T: Config> = StorageValue<_, ScheduleId, ValueQuery>;
+
+	/// Storing schedule details
+	#[pallet::storage]
+	#[pallet::getter(fn schedules)]
+	pub type Schedules<T: Config> =
+		StorageMap<_, Blake2_128Concat, ScheduleId, Schedule<T::AccountId, T::AssetId, BlockNumberFor<T>>, OptionQuery>;
+
+	/// Storing schedule ownership
+	#[pallet::storage]
+	#[pallet::getter(fn owner_of)]
+	pub type ScheduleOwnership<T: Config> =
+		StorageDoubleMap<_, Blake2_128Concat, T::AccountId, Twox64Concat, ScheduleId, (), OptionQuery>;
+
+	/// Keep tracking the remaining amounts to spend for DCA schedules
+	#[pallet::storage]
+	#[pallet::getter(fn remaining_amounts)]
+	pub type RemainingAmounts<T: Config> = StorageMap<_, Blake2_128Concat, ScheduleId, Balance, OptionQuery>;
+
+	/// Keep tracking the retry on error flag for DCA schedules
+	#[pallet::storage]
+	#[pallet::getter(fn retries_on_error)]
+	pub type RetriesOnError<T: Config> = StorageMap<_, Blake2_128Concat, ScheduleId, u8, ValueQuery>;
+
+	/// Keep tracking of the schedule ids to be executed in the block
+	#[pallet::storage]
+	#[pallet::getter(fn schedule_ids_per_block)]
+	pub type ScheduleIdsPerBlock<T: Config> =
+		StorageMap<_, Blake2_128Concat, BlockNumberFor<T>, BoundedVec<ScheduleId, T::MaxSchedulePerBlock>, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T>
+	where
+		<T as pallet_route_executor::Config>::Balance: From<Balance>,
+		Balance: From<<T as pallet_route_executor::Config>::Balance>,
+	{
+		/// Creates a new DCA schedule and plans the execution in the specified start execution block.
+		/// If start execution block number is not specified, then the schedule is planned in the consequent block.
+		///
+		/// The order will be executed within the configured AMM trade pool
+		///
+		/// Parameters:
+		/// - `origin`: schedule owner
+		/// - `schedule`: schedule details
```

Comment:
i think that is key to explain what "schedule" is and order details as well, more in detail is should be explained what is absolute trade limit and dynamic one (slippage), also "total_amout" should be explained in detail and how it behaves in case of sell/buy and fees etc.

Reply:
from looking at this and other docs, we'll probably should completely review & rewrite them later as there is lot work to be done

Reply:
in current state i cannot really imagine anyone being able to use DCA just based on these docs and there is lot of space for interpretation and confusion

Reply:
https://github.com/galacticcouncil/HydraDX-node/issues/595